libimage/runtime.go

```diff
@@ -308,29 +308,53 @@ func (r *Runtime) LookupImage(name string, options *LookupImageOptions) (*Image,
 
 	// The specified name may refer to a short ID. Note that this *must*
 	// happen after the short-name expansion as done above.
-	img, err := r.lookupImageInLocalStorage(name, name, options)
+	img, err := r.lookupImageInLocalStorage(name, name, nil, options)
 	if err != nil {
 		return nil, "", err
 	}
 	if img != nil {
 		return img, name, err
 	}
 
-	return r.lookupImageInDigestsAndRepoTags(name, options)
+	return r.lookupImageInRepoTags(name, possiblyUnqualifiedNamedReference, options)
 }
 
 // lookupImageInLocalStorage looks up the specified candidate for name in the
 // storage and checks whether it's matching the system context.
-func (r *Runtime) lookupImageInLocalStorage(name, candidate string, options *LookupImageOptions) (*Image, error) {
+func (r *Runtime) lookupImageInLocalStorage(name, candidate string, namedCandidate reference.Named, options *LookupImageOptions) (*Image, error) {
 	logrus.Debugf("Trying %q ...", candidate)
-	img, err := r.store.Image(candidate)
-	if err != nil && !errors.Is(err, storage.ErrImageUnknown) {
-		return nil, err
-	}
-	if img == nil {
-		return nil, nil
+
+	var err error
+	var img *storage.Image
+	var ref types.ImageReference
+
+	// for images pulled by tag, Image.Names does not currently contain a
+	// repo@digest value, so such an input would not match.
+	if namedCandidate != nil {
+		ref, err = storageTransport.Transport.NewStoreReference(r.store, namedCandidate, "")
```

Comment:
I think this does need `reference.TagNameOnly(namedCandidate)`, at least I can’t see where else that happens. (Names that match `reference.IsNameOnly` are not stored in `Image.Names` and resolve to no image.)

Reply:
Done :heavy_check_mark: